pydatastructs/miscellaneous_data_structures/stack.py

```diff
@@ -109,3 +113,61 @@ def __str__(self):
         Used for printing.
         """
         return str(self.items._data)
+
+
+class LinkedListStack(Stack):
+
+    def __new__(cls, items=None, dtype=NoneType):
+        obj = object.__new__(cls)
+        obj.stack = SinglyLinkedList()
+        obj._dtype = dtype
+        obj.front = obj.stack.head
+        obj.tail = obj.stack.tail
+        obj.size = 0
+        if items is None:
+            pass
+        elif type(items) in (list, tuple):
+            if len(items) != 0 and dtype is NoneType:
+                obj._dtype = type(items[0])
+            for x in items:
+                if type(x) == obj._dtype:
+                    obj.push(x)
+                else:
+                    raise TypeError("Expected %s but got %s"%(obj._dtype, type(x)))
+        else:
+            raise TypeError("Expected type: list/tuple")
+        return obj
+
+    def push(self, x):
+        if self._dtype is NoneType:
+            self._dtype = type(x)
+        elif type(x) is not self._dtype:
+            raise TypeError("Expected %s but got %s"%(self._dtype, type(x)))
+        self.size += 1
+        self.stack.append_left(x)
+        if self.front is None:
+            self.front = self.stack.head
+        self.rear = self.stack.tail
+
+    def pop(self):
+        if self.is_empty:
+            raise ValueError("Stack is empty")
+        self.size -= 1
+        return_value = self.stack.pop_left()
+        self.front = self.stack.head
+        self.rear = self.stack.tail
+        return return_value
+
+    @property
+    def is_empty(self):
+        return self.size == 0
```

Comment:
Can we use, `self.top` here to figure out whether the stack is empty?

Reply:
It will lose consistency with LinkedListQueue if i change it.

Reply:
It looks like that `LinkedListQueue` needs some changes in it's `__new__` method. Can you please address similar comments there as well.

Reply:
Sure will solve it in different PR.